src/ring/middleware/oauth2.clj

```diff
@@ -79,16 +82,26 @@
 (defn state-mismatch-handler [_]
   {:status 400, :headers {}, :body "State mismatch"})
 
+(defn no-auth-code-handler [_]
+  {:status 400, :headers {}, :body "No authorization code"})
+
 (defn- make-redirect-handler [{:keys [id landing-uri] :as profile}]
-  (let [error-handler (:state-mismatch-handler profile state-mismatch-handler)]
+  (let [state-mismatch-error-handler (:state-mismatch-handler profile state-mismatch-handler)
```

Comment:
Ideally the variable names should match the keys. The line length should also be under 80 characters, and the clauses in the `let` should be aligned.